dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/imports/validation/validator/event/DateValidatorTest.java

```diff
@@ -246,6 +246,24 @@ void testEventIsNotValidWhenDateBelongsToExpiredPeriod() {
     assertHasError(reporter, event, E1047);
   }
 
+  @Test
+  void testEventIsValidWhenScheduledDateBelongsToFuturePeriod() {
```

Comment:
Some time ago, we decided to use the convention for our test names: `should[ExpectedResult]When[Action]`

Reply:
Thanks @muilpp ! Good catch on the expiryDays to 0 meaning indefinite. I'll update the code and add a test case for that scenario as well!

Reply:
@ameenhere, I didn't know how the `exipiryDays` work, so I read our docs, but I couldn't find anything about setting its value to 0, so I asked AI. That said, I'm not sure we should rely on it, wouldn't be the first time it lies. Just wanted to mention it here before we make any decision about it.

Reply:
I do remember expiryDays set to 0 is indeed the default which means "there is no expiry". 

But I was hoping that expiryDays is only (and always) applicable if a "expiryPeriodType" is set to something. 
Meaning if expiryPeriodType is null, then that means there is no expiry (whatever the expiryDays is configured as). 

But if expiryPeriodType is a specific periodType, then expiryDays would be used (even if it is 0) to limit the number days after the end of the period to lock the event. This would also make it possible to actually say, "If the period has ended, lock it!" . This was my assumption. But I realize that may not be existing behaviour.

For resolving the original bug atleast, I think it is safe that I do not change the existing behaviour. Whether or not we should change the behaviour to my assumption, we can decide later. So reverted the if clause change.